### PR TITLE
Rebase & adding support for MatchGroup in Payload

### PR DESCRIPTION
**Pull request by Bernstern**
Most changes are from the rebase, more configuration may be needed since it looks like some more providers are supported now, all changes pertaining to supporting setting the group to match on are in `cmd/root.go`.

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -185,6 +146,17 @@ func initConfig() {
...
+	// ovverride config with fields from the inputConfig
+	if inputConfig.GroupName != "" {
+		cfg.GroupMatch = inputConfig.GroupName
+		log.Infof("Overriding GroupMatch with %s", inputConfig.GroupName)
+	}
+
+	// if inputConfig.UserName != "" {
+	// 	cfg.UserMatch = inputConfig.UserName
+	// 	log.Infof("Overriding UserMatch with %s", inputConfig.UserName)
+	// }
```
Comment (michelleparent) [inline cmd/root.go:158]: do we want to keep this here?

File: cmd/root.go [hunk 2]
```diff
@@ -185,6 +146,17 @@ func initConfig() {
...
+
```
Comment (michelleparent) [inline cmd/root.go:160]: do we want to keep this here?

File: template.yaml [hunk 3]
```diff
@@ -39,13 +50,18 @@ Metadata:
...
+    AllowedPattern: '(?!.*\s)|[a-zA-Z0-9-_]{1,140}'
   ScheduleExpression:
     Type: String
     Description: Schedule for trigger the execution of ssosync (see CloudWatch schedule expressions)
     Default: rate(15 minutes)
+    AllowedPattern: '(?!.*\s)|rate\(\d{1,3} (minutes|hours|days)\)|(cron\((([0-9]|[1-5][0-9]|60)|\d\/([0-9]|[1-5][0-9]|60)|\*) (([0-9]|[1][0-9]|[2][0-3])|(\d\/([0-9]|[1][0-9]|[2][0-3]))|(([0-9]|[1][0-9]|[2][0-3])-([0-9]|[1][0-9]|[2][0-3]))|\*) (([1-9]|[1-2][0-9]|[3][0-1])|\d\/([1-9]|[1-2][0-9]|[3][0-1])|[1-5]W|L|\*|\?) (([1-9]|[1][1-2])|(JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC)|((JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV)-(FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC))|(JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV)(,(FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC)){0,11}|\d\/([0-9]|[1][0-2])|\?|\*) ((MON|TUE|WED|THU|FRI|SAT|SUN)|(MON|TUE|WED|THU|FRI|SAT)-(TUE|WED|THU|FRI|SAT|SUN)|(MON|TUE|WED|THU|FRI|SAT)(,(TUE|WED|THU|FRI|SAT|SUN)){0,6}|[1-7]L|[1-7]#[1-5]|\?|\*) ((19[7-9][0-9]|2[0-1]\d\d)|(19[7-9][0-9]|2[0-1]\d\d)-(19[7-9][0-9]|2[0-1]\d\d)|(19[7-9][0-9]|2[0-1]\d\d)(,(19[7-9][0-9]|2[0-1]\d\d))*|\*)\))'
```
Comment (michelleparent) [inline template.yaml:64]: omg

File: cmd/root.go (hunk 2 above)
Comment (michelleparent) [reply cmd/root.go:160]: oh I see - if you want to leave this here maybe we can add a comment on how to use it

**Review by michelleparent — APPROVED** [review]
lgtm! lol `log.Info("meep")`

File: template.yaml (hunk 3 above)
Comment (Bernstern) [reply template.yaml:64]: Why they don't just use something like cron syntax baffles me to this day 😆

File: cmd/root.go (hunk 2 above)
Comment (Bernstern) [reply cmd/root.go:160]: I can add a comment, wanted to include it incase we ever wanted to get it to the user level, will add a note

**Pull request closed without merging**